### PR TITLE
Task management 1

### PR DESCRIPTION
**Pull request by tanuja009**
Title:Task Management System
Create Models User,Task,Comment
Create views Login,Logout,Signup,Task create,Task Update,Task List, CommentView, DeleteTask

### CODE REVIEW

File: task_management_app/views.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+    return redirect('TaskView')
+  
+class UpdateTask(View):
+  def get(self,request,id):
+    user=request.user
+    task=Task.objects.filter(id=id).values().first()
+    if task:
+      task['start_date'] = task['start_date'].strftime('%Y-%m-%d')
+      task['end_date'] = task['end_date'].strftime('%Y-%m-%d')
+    print(task)
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please remove all the print statements

File: task_management_app/views.py [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+            return redirect('homepage') 
+        else:
+            messages.error(request, "Authentication failed. Please check your credentials.")
+            return redirect("loginform")
+
+
+class signupform(View):
+  def get(self,request):
+    return render(request,'registration.html')
+  def post(self,request):
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please use proper spacing and add docstring. (Follow PEP8)

File: task_management_app/views.py [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+from django.db.models import Q
+from django.dispatch import receiver
+from django.core.mail import send_mail
+from django.conf import settings
+from django.http import HttpResponse
+import datetime
+
+# Create your views here.
+User = get_user_model()
+class loginform(View):
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please follow PEP8 while writing class names

File: task_management_app/views.py [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+    def get(self, request):
+        return render(request, 'login.html')
+
+    def post(self, request):
+        email = request.POST.get('email')
+        password = request.POST.get('Password')
+
+        if not email or not password:
+            messages.error(request, "Email aur password zaruri hai.")
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please use profesional error messages

File: task_management_app/views.py [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+    user=request.user
+    task=Task.objects.filter(id=id).values().first()
+    if task:
+      task['start_date'] = task['start_date'].strftime('%Y-%m-%d')
+      task['end_date'] = task['end_date'].strftime('%Y-%m-%d')
+    print(task)
+    return render(request,'updateform.html',{'task':task})
+  def post(self,request,id):
+    user=request.user
+    task=get_object_or_404(Task,id=id)
```
Comment (sagarjain2104) [inline task_management_app/views.py]: please use try expect

File: task_management_app/urls.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+from django.contrib import admin
+from django.urls import path
+from .views import *
+from django.contrib.auth.decorators import login_required
+
+urlpatterns = [
+   path('',loginform.as_view(),name="loginform"),
+   path('signupform/',signupform.as_view(),name="signupform"),
+   path('homepage/',login_required(Homepage.as_view()),name="homepage"),
```
Comment (sagarjain2104) [inline task_management_app/urls.py]: Please use login_required in views

File: task_management_app/models.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+from django.db import models
+from django.contrib.auth.models import AbstractUser
+from django.utils.timezone import now
+from .manager import CustomUserManager
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth.models import Group,Permission
+
+
+
+class Timestampmodel(models.Model):
```
Comment (sagarjain2104) [inline task_management_app/models.py]: Django also provides TimeStamp model find and replace it

File: task_management_app/templates/login.html [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      <div class="pass"><a href="#">Forgot password?</a></div>
+      <div class="row button">
+        <input type="submit" value="Login" />
+      </div>
+      <div class="signup-link">Not a member? <a href="{% url 'signupform'%}">Signup now</a></div>
+    </form>
+  </div>
+</body>
+</html>
+
```
Comment (sagarjain2104) [inline task_management_app/templates/login.html:50]: Don't push extra spaces

File: task_management_app/views.py [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+      return render(request,'commentshow.html',{'comment':comment})
+    else:
+      return render(request,'commentshow.html',{'comments':comments})
+    
+class Usercreate(View):
+  def get(self,request):
+    return render(request,'usercreate.html')
+  
+  def post(self,request):
+    user=request.user
```
Comment (sagarjain2104) [inline task_management_app/views.py]: user variable is not used

File: task_management_app/views.py [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+    user=User.objects.all()
+    email=request.POST.get('email')
+    phone_no=request.POST.get('phone')
+    User.objects.create_user(email=email,phone_no=phone_no)
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please use forms and validations

**Comment by tanishgupta905** [comment]
fetch latest changes from main branch and merge in this PR

File: task_management_app/views.py (hunk 2 above)
Comment (tanishgupta905) [reply task_management_app/views.py]: and reformat code via Black

File: task_management_app/views.py [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+            return redirect("homepage")
+        else:
+            messages.error(
+                request,
+                "Authentication failed. Please check your credentials.",
+            )
+            return redirect("loginform")
+
+
+class SignupForm(View):
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please do not use Form in view name, (use SignUpView)

File: task_management_app/urls.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+urlpatterns = [
+    path("", LoginForm.as_view(), name="loginform"),
+    path("signupform/", SignupForm.as_view(), name="signupform"),
+    path("homepage/", HomePage.as_view(), name="homepage"),
+    # path('userhome/',userhome.as_view(),name="userhome"),
+    path("logoutpage/", LogoutPage.as_view(), name="logoutpage"),
+    path("Profileview/", ProfileView.as_view(), name="profileview"),
+    path("TaskCreate/", TaskCreate.as_view(), name="TaskCreate"),
+    path("TaskView/", TaskView.as_view(), name="TaskView"),
+    path("Taskdetails/<int:id>", TaskDetails.as_view(), name="Taskdetails"),
```
Comment (sagarjain2104) [inline task_management_app/urls.py]: please do not use capital url names and try to improve urls like task-details or task/details/

File: task_management_app/views.py [hunk 13]
```diff
@@ -0,0 +1,297 @@
...
+    def get(self, request):
+        return render(request, "registration.html")
+
+    def post(self, request):
+        email = request.POST.get("email")
```
Comment (sagarjain2104) [inline task_management_app/views.py]: Please use django forms and add validations there

**Review by sagarjain2104 — APPROVED** [review]

**Pull request merged**